src/switchroot/ostree-prepare-root.c

```diff
@@ -221,16 +220,6 @@ main(int argc, char *argv[])
 
   deploy_path = resolve_deploy_path (root_mountpoint);
 
-  /* Create a temporary target for our mounts in the initramfs; this will
-   * be moved to the new system root below.
```

Comment:
This code was written for a reason in 4f75d4ea0b640dd69f13fd31976d6ddfb42c977a - are you sure it won't reintroduce that bug?